core/src/main/scala/org/apache/spark/network/BlockTransferService.scala

```diff
@@ -83,7 +83,12 @@ abstract class BlockTransferService extends ShuffleClient with Closeable with Lo
    *
    * It is also only available after [[init]] is invoked.
    */
-  def fetchBlockSync(host: String, port: Int, execId: String, blockId: String): ManagedBuffer = {
+  def fetchBlockSync(
+      host: String,
+      port: Int,
+      execId: String,
+      blockId: String,
+      timeout: Int): ManagedBuffer = {
```

Comment:
Maybe we should name this parameter `timeoutSeconds` to minimize any chance for confusion.